src/functions-and-arrays.js

```diff
@@ -1,19 +1,139 @@
 // Iteration #1: Find the maximum
+const maxOfTwoNumbers = (a,b) => {
+  if (a>b) {
+    return a;
+  } else if (a<b) {
+    return b;
+  } else {
+    return a;
+  }
+};
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
+const findLongestWord = (array) => {
+  if (array.length === 0) {
+    return null;
+  }
+
+  let longestWord = array[0];
+  for (let i = 0; i < array.length; i++) {
+    
+    if (array[i].length > longestWord.length) {
+      longestWord = array[i];
+    }
+  };
+  return longestWord;
+};
+
+//console.log(findLongestWord(words));
+
+
 // Iteration #3: Calculate the sum
 
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+const sumNumbers = (array) => {
+  if (array.length === 0) {
+    return 0;
+  }
+  
+  let sum = 0;
+
+  for (let i = 0; i < array.length; i++) {
+    sum += array[i];
+  };
+  
+  return sum;
+};
+
+//console.log(sumNumbers(numbers));
+
+//Iteration #3.1: Bonus
+
+const mixedArr = [6, 12, 'miami', 1, true, 'barca', '200', 'lisboa', 8, 10];
+
+const sum = (array) => {
+  
+  let total = 0;
+
+  //if (typeof array[i] === "object") {
+  //  throw new Error("unsupported");
+  //}  
+
+  for (let i = 0; i < array.length; i++) {
+
+
+    switch (typeof array[i]) {
+      case "number":
+        total+= array[i];
+        break;
+      case "string":
+        total += array[i].length;
+        break;
+      case "boolean":
+        if (array[i] === true) {
+          total += 1;
+        } else {
+          continue;
+        };
+        break;
+      default:
```

Comment:
To pass the test in Jasmine you couldn't pass you could just write something like this here instead of the "break;":

throw new Error("Unsupported data type sir or ma'am");